src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CertificateAssetDownloader.cs

```diff
@@ -102,68 +105,91 @@ internal static class CertificateAssetDownloader
+        [DynamicDependency("Send", "System.Net.Http.HttpClient", "System.Net.Http")]
+        [DynamicDependency("Content", "System.Net.Http.HttpResponseMessage", "System.Net.Http")]
+        [DynamicDependency("ReadAsStream", "System.Net.Http.HttpContent", "System.Net.Http")]
+        private static Func<string, CancellationToken, byte[]>? CreateDownloadBytesFunc()
         {
             try
             {
                 // Use reflection to access System.Net.Http:
                 // Since System.Net.Http.dll explicitly depends on System.Security.Cryptography.X509Certificates.dll,
                 // the latter can't in turn have an explicit dependency on the former.
 
+                // Get the relevant types needed.
                 Type? socketsHttpHandlerType = Type.GetType("System.Net.Http.SocketsHttpHandler, System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: false);
                 Type? httpClientType = Type.GetType("System.Net.Http.HttpClient, System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: false);
-                if (socketsHttpHandlerType == null || httpClientType == null)
+                Type? httpRequestMessageType = Type.GetType("System.Net.Http.HttpRequestMessage, System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: false);
+                Type? httpResponseMessageType = Type.GetType("System.Net.Http.HttpResponseMessage, System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: false);
+                Type? httpContentType = Type.GetType("System.Net.Http.HttpContent, System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: false);
+                if (socketsHttpHandlerType == null || httpClientType == null || httpRequestMessageType == null || httpResponseMessageType == null || httpContentType == null)
                 {
+                    Debug.Fail("Unable to load required type.");
                     return null;
                 }
 
+                // Get the methods on those types.
                 PropertyInfo? pooledConnectionIdleTimeoutProp = socketsHttpHandlerType.GetProperty("PooledConnectionIdleTimeout");
-                MethodInfo? getByteArrayAsyncMethod = httpClientType.GetMethod("GetByteArrayAsync", new Type[] { typeof(string) });
-                if (pooledConnectionIdleTimeoutProp == null || getByteArrayAsyncMethod == null)
+                PropertyInfo? requestUriProp = httpRequestMessageType.GetProperty("RequestUri");
+                MethodInfo? sendMethod = httpClientType.GetMethod("Send", new Type[] { httpRequestMessageType, typeof(CancellationToken) });
+                PropertyInfo? responseContentProp = httpResponseMessageType.GetProperty("Content");
+                MethodInfo? readAsStreamAsyncMethod = httpContentType.GetMethod("ReadAsStream", Type.EmptyTypes);
```

Comment:
Shouldn't it be `readAsStreamMethod` instead of `readAsStreamAsyncMethod`?

Reply:
Yes, I fixed up the usage to use the ReadAsStream you merged this morning, but I neglected to rename the local. Done.